File: app/src/main/java/com/brainwellnessspa/Utility/MyNetworkReceiver.java
package com.brainwellnessspa.Utility;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Handler;
import android.os.Looper;
import android.util.Log;

import com.brainwellnessspa.BWSApplication;
import com.brainwellnessspa.EncryptDecryptUtils.DownloadMedia;
import com.downloader.PRDownloader;
import com.downloader.Status;
import com.google.android.exoplayer2.ExoPlayer;
import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;

import java.lang.reflect.Type;
import java.util.List;

import static com.brainwellnessspa.DashboardModule.Activities.AudioPlayerActivity.AudioInterrupted;
import static com.brainwellnessspa.EncryptDecryptUtils.DownloadMedia.downloadIdOne;
import static com.brainwellnessspa.EncryptDecryptUtils.DownloadMedia.isDownloading;
import static com.brainwellnessspa.Services.GlobalInitExoPlayer.player;

public class MyNetworkReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(final Context context, final Intent intent) {
        boolean status = BWSApplication.isNetworkConnected(context);

        Log.d("network",String.valueOf(status));
        if(!status) {
            if (isDownloading) {
                PRDownloader.pause(downloadIdOne);
                isDownloading = false;
//                BWSApplication.showToast(String.valueOf(status)+Status.valueOf(PRDownloader.getStatus(downloadIdOne).name()),context);
            }
        }else {
            if(player!=null){
//                for(int i = 0;i<5;i++) {
                    if (player.getPlaybackState() == ExoPlayer.STATE_IDLE && AudioInterrupted) {
                        AudioInterrupted = false;
                        player.setPlayWhenReady(true);
                        player.prepare();
                        player.seekTo(player.getCurrentPosition());
                        Log.e("Exo PLayer Net:", "Player Resume after Net");
                    }
//                }
            }
            if (!isDownloading) {
                List<String> fileNameList, audioFile, playlistDownloadId;
                if (BWSApplication.isNetworkConnected(context)) {
                    SharedPreferences sharedx = context.getSharedPreferences(CONSTANTS.PREF_KEY_DownloadPlaylist, Context.MODE_PRIVATE);
                    Gson gson = new Gson();
                    String json = sharedx.getString(CONSTANTS.PREF_KEY_DownloadName, String.valueOf(gson));
                    String json1 = sharedx.getString(CONSTANTS.PREF_KEY_DownloadUrl, String.valueOf(gson));
                    String json2 = sharedx.getString(CONSTANTS.PREF_KEY_DownloadPlaylistId, String.valueOf(gson));
                    if (!json1.equalsIgnoreCase(String.valueOf(gson))) {
                        Type type = new TypeToken<List<String>>() {
                        }.getType();
                        fileNameList = gson.fromJson(json, type);
                        audioFile = gson.fromJson(json1, type);
                        playlistDownloadId = gson.fromJson(json2, type);
                   /* fileNameList = new ArrayList<>();
                    audioFile = new ArrayList<>();
                    playlistDownloadId = new ArrayList<>();
                    SharedPreferences sharedxxx = getActivity().getSharedPreferences(CONSTANTS.PREF_KEY_DownloadPlaylist, Context.MODE_PRIVATE);
                    SharedPreferences.Editor editor = sharedxxx.edit();
                    String nameJson = gson.toJson(fileNameList);
                    String urlJson = gson.toJson(audioFile);
                    String playlistIdJson = gson.toJson(playlistDownloadId);
                    editor.putString(CONSTANTS.PREF_KEY_DownloadName, nameJson);
                    editor.putString(CONSTANTS.PREF_KEY_DownloadUrl, urlJson);
                    editor.putString(CONSTANTS.PREF_KEY_DownloadPlaylistId, playlistIdJson);
                    editor.commit();*/
                        if (fileNameList.size() != 0) {
                            isDownloading = true;
                            DownloadMedia downloadMedia = new DownloadMedia(context.getApplicationContext());
                            downloadMedia.encrypt1(audioFile, fileNameList, playlistDownloadId/*, playlistSongs*/);
                        }
                    }
                }
            }
        }
    }
}

File: app/src/main/java/com/brainwellnessspa/utility/MyNetworkReceiver.kt
package com.brainwellnessspa.utility

import android.annotation.SuppressLint
import android.app.Activity
import android.app.Service
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log
import com.brainwellnessspa.BWSApplication
import com.brainwellnessspa.encryptDecryptUtils.DownloadMedia
import com.brainwellnessspa.services.GlobalInitExoPlayer
import com.downloader.PRDownloader
import com.google.gson.Gson
import com.google.gson.reflect.TypeToken
import java.util.*

class MyNetworkReceiver : BroadcastReceiver() {

    @SuppressLint("UnsafeProtectedBroadcastReceiver")
    override fun onReceive(context: Context, intent: Intent) {
        val status = BWSApplication.isNetworkConnected(context)
        Log.d("network", status.toString())
        if (!status) {
            if (DownloadMedia.isDownloading) {
                PRDownloader.pause(DownloadMedia.downloadIdOne)
                DownloadMedia.isDownloading = false //                BWSApplication.showToast(String.valueOf(status)+Status.valueOf(PRDownloader.getStatus(downloadIdOne).name()),context);
            }
        } else {
            val shareded = context.getSharedPreferences(CONSTANTS.PREF_KEY_USER_ACTIVITY, Service.MODE_PRIVATE)
            val gson = Gson()
            val json = shareded.getString(CONSTANTS.PREF_KEY_USER_TRACK_ARRAY, gson.toString())
            var userActivityTrackModel = ArrayList<UserActivityTrackModel>()
            if (!json.equals(gson.toString(), ignoreCase = true)) {
                val type = object : TypeToken<ArrayList<UserActivityTrackModel?>?>() {}.type
                userActivityTrackModel = gson.fromJson(json, type)
                if (userActivityTrackModel.size != 0) {
                    val global = GlobalInitExoPlayer()
                    global.getUserActivityCall(context, "", "", "")
                }
            }
// TODO check shared pref and aema jo value hoy to get karine api call karvani
            // do some thing
            GlobalInitExoPlayer.callResumePlayer(context)
            if (!DownloadMedia.isDownloading) {
                val fileNameList: List<String>
                val audioFile: List<String>
                val playlistDownloadId: List<String>
                if (BWSApplication.isNetworkConnected(context)) {
                    val shared = context.getSharedPreferences(CONSTANTS.PREF_KEY_DownloadPlaylist, Context.MODE_PRIVATE)
                    val gson = Gson()
                    val json = shared.getString(CONSTANTS.PREF_KEY_DownloadName, gson.toString())
                    val json1 = shared.getString(CONSTANTS.PREF_KEY_DownloadUrl, gson.toString())
                    val json2 = shared.getString(CONSTANTS.PREF_KEY_DownloadPlaylistId, gson.toString())
                    if (!json1.equals(gson.toString(), ignoreCase = true)) {
                        val type = object : TypeToken<List<String?>?>() {}.type
                        fileNameList = gson.fromJson(json, type)
                        audioFile = gson.fromJson(json1, type)
                        playlistDownloadId = gson.fromJson(json2, type)
                        /* fileNameList = new ArrayList<>();
                    audioFile = new ArrayList<>();
                    playlistDownloadId = new ArrayList<>();
                    SharedPreferences sharedxxx = getActivity().getSharedPreferences(CONSTANTS.PREF_KEY_DownloadPlaylist, Context.MODE_PRIVATE);
                    SharedPreferences.Editor editor = sharedxxx.edit();
                    String nameJson = gson.toJson(fileNameList);
                    String urlJson = gson.toJson(audioFile);
                    String playlistIdJson = gson.toJson(playlistDownloadId);
                    editor.putString(CONSTANTS.PREF_KEY_DownloadName, nameJson);
                    editor.putString(CONSTANTS.PREF_KEY_DownloadUrl, urlJson);
                    editor.putString(CONSTANTS.PREF_KEY_DownloadPlaylistId, playlistIdJson);
                    editor.commit();*/
                        if (fileNameList.isNotEmpty()) {
                            DownloadMedia.isDownloading = true
                            val downloadMedia = DownloadMedia(context.applicationContext, context as Activity)
                            downloadMedia.encrypt1(audioFile, fileNameList, playlistDownloadId /*, playlistSongs*/)
                        }
                    }
                }
            }
        }
    }
}